### PR TITLE
Add the virtio support tiling buffer format

### PR DESCRIPTION
**Pull request by yhe39**
Tracked-On: OAM-110757

### CODE REVIEW

File: bsp_diff/celadon_ivi/kernel/linux-intel-lts2021/0003-Add-support-for-virtgpu_gem_prime_import_sg_table.patch [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+--- a/drivers/gpu/drm/virtio/virtgpu_drv.h
++++ b/drivers/gpu/drm/virtio/virtgpu_drv.h
+@@ -454,16 +454,35 @@ void virtio_gpu_fence_event_process(struct virtio_gpu_device *vdev,
+ void virtio_gpu_cleanup_object(struct virtio_gpu_object *bo);
+ struct drm_gem_object *virtio_gpu_create_object(struct drm_device *dev,
+ 						size_t size);
++//int virtio_gpu_object_create(struct virtio_gpu_device *vgdev,
++//			     struct virtio_gpu_object_params *params,
++//			     struct virtio_gpu_object **bo_ptr,
++//			     struct virtio_gpu_fence *fence);
```
Comment (feijiang1) [inline bsp_diff/celadon_ivi/kernel/linux-intel-lts2021/0003-Add-support-for-virtgpu_gem_prime_import_sg_table.patch:29]: should remove useless code

File: bsp_diff/celadon_ivi/kernel/linux-intel-lts2021/0002-Add-support-for-modifier-in-virtio-gpu.patch [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+diff --git a/drivers/gpu/drm/virtio/virtgpu_display.c b/drivers/gpu/drm/virtio/virtgpu_display.c
+index 2df9d839ecad..4b0a52135f8e 100644
+--- a/drivers/gpu/drm/virtio/virtgpu_display.c
++++ b/drivers/gpu/drm/virtio/virtgpu_display.c
+@@ -344,7 +344,7 @@ int virtio_gpu_modeset_init(struct virtio_gpu_device *vgdev)
+ 	vgdev->ddev->mode_config.max_width = XRES_MAX;
+ 	vgdev->ddev->mode_config.max_height = YRES_MAX;
+ 
+-	vgdev->ddev->mode_config.fb_modifiers_not_supported = true;
++//	vgdev->ddev->mode_config.fb_modifiers_not_supported = true;
```
Comment (feijiang1) [inline bsp_diff/celadon_ivi/kernel/linux-intel-lts2021/0002-Add-support-for-modifier-in-virtio-gpu.patch:34]: remove useless code

File: bsp_diff/celadon_ivi/kernel/linux-intel-lts2021/0002-Add-support-for-modifier-in-virtio-gpu.patch [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+diff --git a/drivers/gpu/drm/drm_modeset_helper.c b/drivers/gpu/drm/drm_modeset_helper.c
+index f858dfedf2cf..279bdb902798 100644
+--- a/drivers/gpu/drm/drm_modeset_helper.c
++++ b/drivers/gpu/drm/drm_modeset_helper.c
+@@ -93,6 +93,7 @@ void drm_helper_mode_fill_fb_struct(struct drm_device *dev,
+ 	}
+ 	fb->modifier = mode_cmd->modifier[0];
+ 	fb->flags = mode_cmd->flags;
++	printk("drm_helper_mode_fill_fb_struct : modifier = %llx in virtio_gpu_user_framebuffer_create ",fb->modifier);
```
Comment (feijiang1) [inline bsp_diff/celadon_ivi/kernel/linux-intel-lts2021/0002-Add-support-for-modifier-in-virtio-gpu.patch:21]: should not use printk, if need add log, please use DRM_DEBUG_***

File: bsp_diff/celadon_ivi/kernel/linux-intel-lts2021/0002-Add-support-for-modifier-in-virtio-gpu.patch [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+@@ -112,12 +112,28 @@ uint32_t virtio_gpu_translate_format(uint32_t drm_fourcc)
+ 	return format;
+ }
+ 
++static bool virtio_gpu_plane_format_mod_supported(struct drm_plane *_plane,
++					    u32 format, u64 modifier)
++{
++	switch (modifier) {
++
++	printk("format mode supported  modifier = %llx ",modifier);
```
Comment (feijiang1) [inline bsp_diff/celadon_ivi/kernel/linux-intel-lts2021/0002-Add-support-for-modifier-in-virtio-gpu.patch:51]: should not use printk, if need add log, please use DRM_DEBUG_***

File: bsp_diff/celadon_ivi/kernel/linux-intel-lts2021/0002-Add-support-for-modifier-in-virtio-gpu.patch (hunk 3 above)
Comment (feijiang1) [inline bsp_diff/celadon_ivi/kernel/linux-intel-lts2021/0002-Add-support-for-modifier-in-virtio-gpu.patch:21]: should not use printk, if need add log, please use DRM_DEBUG_***

File: bsp_diff/celadon_ivi/kernel/linux-intel-lts2021/0003-Add-support-for-virtgpu_gem_prime_import_sg_table.patch [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
++       } else {
++               for_each_sgtable_sg(table, sg, si) {
++                       (*ents)[si].addr = cpu_to_le64(sg_phys(sg));
++                       (*ents)[si].length = cpu_to_le32(sg->length);
++                       (*ents)[si].padding = 0;
++               }
++       }
++
++       return 0;
++
```
Comment (feijiang1) [inline bsp_diff/celadon_ivi/kernel/linux-intel-lts2021/0003-Add-support-for-virtgpu_gem_prime_import_sg_table.patch:184]: empty line remove

File: bsp_diff/celadon_ivi/kernel/linux-intel-lts2021/0003-Add-support-for-virtgpu_gem_prime_import_sg_table.patch [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+@@ -162,9 +162,93 @@ struct drm_gem_object *virtgpu_gem_prime_import(struct drm_device *dev,
+ 	return drm_gem_prime_import(dev, buf);
+ }
+ 
++int virtio_gpu_sgt_to_mem_entry(struct virtio_gpu_device *vgdev,
++                                 struct sg_table *table,
++                                  struct virtio_gpu_mem_entry **ents,
++                                 unsigned int *nents)
++{
++
```
Comment (feijiang1) [inline bsp_diff/celadon_ivi/kernel/linux-intel-lts2021/0003-Add-support-for-virtgpu_gem_prime_import_sg_table.patch:151]: empty line remove

File: bsp_diff/celadon_ivi/kernel/linux-intel-lts2021/0002-Add-support-for-modifier-in-virtio-gpu.patch (hunk 3 above)
Comment (yhe39) [reply bsp_diff/celadon_ivi/kernel/linux-intel-lts2021/0002-Add-support-for-modifier-in-virtio-gpu.patch]: Hang said it is only for debug, could remove it.

Comment (yhe39) [reply bsp_diff/celadon_ivi/kernel/linux-intel-lts2021/0002-Add-support-for-modifier-in-virtio-gpu.patch]: remove it per Hang's suggestion.

**Review by feijiang1 — APPROVED** [review]
LGTM

**Review by JeevakaPrabu — APPROVED** [review]

**Review by swaroopbalan — APPROVED** [review]

**Review by feijiang1 — APPROVED** [review]
LGTM

**Pull request closed without merging**